packages/fxa-auth-server/test/local/routes/subscriptions/stripe-webhooks.js

```diff
@@ -486,6 +517,78 @@ describe('StripeWebhookHandler', () => {
       });
     });
 
+    describe('handleProductUpdatedEvent', () => {
+      let scopeContextSpy, scopeSpy;
+      beforeEach(() => {
+        scopeContextSpy = sinon.fake();
+        scopeSpy = {
+          setContext: scopeContextSpy,
+        };
+        sandbox.replace(Sentry, 'withScope', (fn) => fn(scopeSpy));
+      });
+
+      it('throws a sentry error if the update event data is invalid', async () => {
+        const updatedEvent = deepCopy(eventProductUpdated);
+        await StripeWebhookHandlerInstance.handleProductUpdatedEvent(
+          {},
+          updatedEvent
+        );
+
+        assert.isTrue(scopeContextSpy.called, 'Expected to call Sentry');
+      });
+
+      it('does not throw a sentry error if the update event data is valid', async () => {
+        const updatedEvent = deepCopy(eventProductUpdated);
+        updatedEvent.data.object.metadata[
+          'product:termsOfServiceDownloadURL:en-us'
+        ] = 'https://accounts-static.cdn.mozilla.net/legal/mozilla_vpn_tos';
+        await StripeWebhookHandlerInstance.handleProductUpdatedEvent(
+          {},
+          updatedEvent
+        );
+
+        assert.isTrue(scopeContextSpy.notCalled, 'Expected not to call Sentry');
+      });
+    });
+
+    describe('handlePlanUpdatedEvent', () => {
+      let scopeContextSpy, scopeSpy;
+      beforeEach(() => {
+        scopeContextSpy = sinon.fake();
+        scopeSpy = {
+          setContext: scopeContextSpy,
+        };
+        sandbox.replace(Sentry, 'withScope', (fn) => fn(scopeSpy));
+      });
+
+      it('throws a sentry error if the update event data is invalid', async () => {
+        const updatedEvent = deepCopy(eventPlanUpdated);
+        updatedEvent.data.object.metadata = {
+          'product:termsOfServiceDownloadURL:en-us':
+            'https://FAIL.net/legal/mozilla_vpn_tos',
+        };
+        await StripeWebhookHandlerInstance.handlePlanUpdatedEvent(
+          {},
+          updatedEvent
+        );
+
+        assert.isTrue(scopeContextSpy.called, 'Expected to call Sentry');
+      });
+
+      it('does not throw a sentry error if the update event data is valid', async () => {
+        const updatedEvent = deepCopy(eventPlanUpdated);
+        updatedEvent.data.object.metadata[
+          'product:termsOfServiceDownloadURL:en-us'
+        ] = 'https://accounts-static.cdn.mozilla.net/legal/mozilla_vpn_tos';
+        await StripeWebhookHandlerInstance.handlePlanUpdatedEvent(
+          {},
+          updatedEvent
+        );
+
+        assert.isTrue(scopeContextSpy.notCalled, 'Expected not to call Sentry');
+      });
```

Comment:
There's an early return when the product is not found.  I don't see that tested in these tests.